pennylane/transforms/decompositions/clifford_plus_t/solovay_kitaev.py

```diff
@@ -0,0 +1,460 @@
+def sk_decomposition(op, depth, basis_set=(), basis_depth=10, approximate_set=None, kd_tree=None):
+    r"""Approximate an arbitrary single-qubit gate in the Clifford+T basis using the `Solovay-Kitaev algorithm <https://arxiv.org/abs/quant-ph/0505030>`_.
+
+    This method implements a recursive Solovay-Kitaev decomposition that approximates any :math:`U \in \text{SU}(2)`
+    operation with :math:`\epsilon > 0` error. The error depends on the recursion ``depth``. In general, this
+    algorithm runs in :math:`O(\text{log}^{2.71}(1/\epsilon))` time and produces a decomposition with
+    :math:`O(\text{log}^{3.97}(1/\epsilon))` operations.
+
+    Args:
+        op (~.Operation): A single-qubit gate operation
+        depth (int): Depth until which the recursion occurs
+        basis_set (list(str)): Basis set to be used for the decomposition and building the ``approximate_set``. It
+            accepts the following gate terms: ``['x', 'y', 'z', 'h', 't', 'tdg', 's', 'sdg']``, where `dg` refers
+            to the gate adjoint. Default value is ``['h', 't', 'tdg']``
+        basis_depth (int): Maximum expansion length of Clifford+T sequences in the ``approximate_set``. Default is `10`
+        approximate_set (list): A list of gate sequences that are used to find an approximation of unitaries in the
+            base case of recursion. This can be precomputed using the :func:`~.sk_approximate_set`
+            method, otherwise will be built using the default/given values of ``basis_set`` and ``basis_depth``
+        kd_tree (scipy.spatial.KDTree): A KD-Tree corresponding to the ``approximate_set``. This can also be
+            precomputed using :func:`~.sk_approximate_set`, otherwise will be built internally.
+
+    Returns:
+        list(~.Operations): A list of gates in the Clifford+T basis set that approximates the given operation
+
+    **Example**
+
+    Suppose one would like to decompose :class:`~.RZ` with :math:`\phi = \pi/3`:
+
+    .. code-block:: python3
+
+        import functools
+        import numpy as np
+        import pennylane as qml
+
+        op  = qml.RZ(np.pi/3, wires=0)
+
+        # Get the gate decomposition in ['t', 'tdg', 'h']
+        ops = qml.transforms.decompositions.sk_decomposition(op, depth=2)
+
+        # Get SU2 matrix from the ops
+        op_matrix = functools.reduce(lambda x, y: x @ y, map(qml.matrix, gates))
+        su2_matrix = op_matrix / np.sqrt((1 + 0j) * np.linalg.det(op_matrix))
+
+    When the function is run for a sufficient ``depth`` with a good enough ``approximate_set``,
+    the output gate sequence should implement the same operation approximately.
+
+    >>> qml.math.allclose(op.matrix(), su2_matrix, atol=1e-3)
+    True
+
+    .. seealso:: :func:`~.sk_approximate_set` for precomputing the ``approximate_set`` and ``kd_tree``.
+    """
+    with QueuingManager.stop_recording():
+        # Check for length of wires in the operation
+        if len(op.wires) > 1:
```

Comment:
in case of globalphase, snapshot, etc.
```suggestion
        if len(op.wires) != 1:
```